app/Services/ImageGenerationService.php

```
@@ -23,72 +28,82 @@ public static function generateImage(string $markup, $deviceId): string
                     ->setOption('args', config('app.puppeteer_docker') ? ['--no-sandbox', '--disable-setuid-sandbox', '--disable-gpu'] : [])
-                    ->windowSize(800, 480)
-                    ->save($pngPath);
-            } catch (\Exception $e) {
-                \Log::error('Failed to generate PNG: '.$e->getMessage());
-                throw new \RuntimeException('Failed to generate PNG: '.$e->getMessage(), 0, $e);
+                    ->windowSize(800, 480);
+
+                if (config('app.puppeteer_wait_for_network_idle')) {
+                    $browsershot->waitUntilNetworkIdle();
+                }
+                ray($browsershot);
```

PHPStan (2.1.15)

Function ray not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(function.notFound)